httpcore5/src/main/java/org/apache/hc/core5/http/ssl/TlsCiphers.java

```diff
@@ -0,0 +1,374 @@
+                    "TLS_PSK_WITH_ARIA_128_GCM_SHA256",
+                    "TLS_PSK_WITH_ARIA_256_GCM_SHA384",
+                    "TLS_RSA_PSK_WITH_ARIA_128_GCM_SHA256",
+                    "TLS_RSA_PSK_WITH_ARIA_256_GCM_SHA384",
+                    "TLS_ECDHE_PSK_WITH_ARIA_128_CBC_SHA256",
+                    "TLS_ECDHE_PSK_WITH_ARIA_256_CBC_SHA384",
+                    "TLS_ECDHE_ECDSA_WITH_CAMELLIA_128_CBC_SHA256",
+                    "TLS_ECDHE_ECDSA_WITH_CAMELLIA_256_CBC_SHA384",
+                    "TLS_ECDH_ECDSA_WITH_CAMELLIA_128_CBC_SHA256",
+                    "TLS_ECDH_ECDSA_WITH_CAMELLIA_256_CBC_SHA384",
+                    "TLS_ECDHE_RSA_WITH_CAMELLIA_128_CBC_SHA256",
+                    "TLS_ECDHE_RSA_WITH_CAMELLIA_256_CBC_SHA384",
+                    "TLS_ECDH_RSA_WITH_CAMELLIA_128_CBC_SHA256",
+                    "TLS_ECDH_RSA_WITH_CAMELLIA_256_CBC_SHA384",
+                    "TLS_RSA_WITH_CAMELLIA_128_GCM_SHA256",
+                    "TLS_RSA_WITH_CAMELLIA_256_GCM_SHA384",
+                    "TLS_DH_RSA_WITH_CAMELLIA_128_GCM_SHA256",
+                    "TLS_DH_RSA_WITH_CAMELLIA_256_GCM_SHA384",
+                    "TLS_DH_DSS_WITH_CAMELLIA_128_GCM_SHA256",
+                    "TLS_DH_DSS_WITH_CAMELLIA_256_GCM_SHA384",
+                    "TLS_DH_anon_WITH_CAMELLIA_128_GCM_SHA256",
+                    "TLS_DH_anon_WITH_CAMELLIA_256_GCM_SHA384",
+                    "TLS_ECDH_ECDSA_WITH_CAMELLIA_128_GCM_SHA256",
+                    "TLS_ECDH_ECDSA_WITH_CAMELLIA_256_GCM_SHA384",
+                    "TLS_ECDH_RSA_WITH_CAMELLIA_128_GCM_SHA256",
+                    "TLS_ECDH_RSA_WITH_CAMELLIA_256_GCM_SHA384",
+                    "TLS_PSK_WITH_CAMELLIA_128_GCM_SHA256",
+                    "TLS_PSK_WITH_CAMELLIA_256_GCM_SHA384",
+                    "TLS_RSA_PSK_WITH_CAMELLIA_128_GCM_SHA256",
+                    "TLS_RSA_PSK_WITH_CAMELLIA_256_GCM_SHA384",
+                    "TLS_PSK_WITH_CAMELLIA_128_CBC_SHA256",
+                    "TLS_PSK_WITH_CAMELLIA_256_CBC_SHA384",
+                    "TLS_DHE_PSK_WITH_CAMELLIA_128_CBC_SHA256",
+                    "TLS_DHE_PSK_WITH_CAMELLIA_256_CBC_SHA384",
+                    "TLS_RSA_PSK_WITH_CAMELLIA_128_CBC_SHA256",
+                    "TLS_RSA_PSK_WITH_CAMELLIA_256_CBC_SHA384",
+                    "TLS_ECDHE_PSK_WITH_CAMELLIA_128_CBC_SHA256",
+                    "TLS_ECDHE_PSK_WITH_CAMELLIA_256_CBC_SHA384",
+                    "TLS_RSA_WITH_AES_128_CCM",
+                    "TLS_RSA_WITH_AES_256_CCM",
+                    "TLS_RSA_WITH_AES_128_CCM_8",
+                    "TLS_RSA_WITH_AES_256_CCM_8",
+                    "TLS_PSK_WITH_AES_128_CCM",
+                    "TLS_PSK_WITH_AES_256_CCM",
+                    "TLS_PSK_WITH_AES_128_CCM_8",
+                    "TLS_PSK_WITH_AES_256_CCM_8"
+            )));
+
+    public static boolean isH2Blacklisted(final String cipherSuite) {
+        return H2_BLACKLISTED.contains(cipherSuite);
```

Comment:
Should the method use normalized strings?

Reply:
@artem-smotrakov Not sure how as those cipher names may contain mixed-case letters.

Reply:
The JSSE docs and RFCs mention `_anon_` ciphers in lower case only but teoretically some TLS providers can use upper case. The method can expect it. That's also what you requested to change in https://github.com/apache/httpcomponents-client/pull/140#pullrequestreview-209877810

Reply:
@artem-smotrakov I remember that. I find apache/httpcomponents-client#140 to be a different story given that it employs pattern matching to filter out weak cipher. Here we have a list of specific ciphers.